### DISCUSSION TITLE
Wifibroadcast NG via Nexmon - programm tries to open nonexisting serial port

### DISCUSSION THREAD

**Discussion by Joshy-2010** [svpcom/wfb-ng#269] [Questions]
Nexmon basically builds a patched kernel module for the internal wifi chip of the raspberry Pi to support hidden features.

I tried building wfb-ng and [Nexmon](https://github.com/seemoo-lab/nexmon) from source. Both seemed to install well. I can confirm that via the patched nexmon driver the wificard now supports both monitor mode as well as injection.

I followed everything pointed out in https://github.com/svpcom/wfb-ng/wiki/Setup-HOWTO and successfully activated the wifibroadcast@drone service. All seems well. Sadly, the service crashes after a short while. The log reveals something curious:

Jan 14 23:25:13 raspberrypi systemd[1]: wifibroadcast@drone.service: Service RestartSec=5s expired, scheduling restart.
Jan 14 23:25:13 raspberrypi systemd[1]: wifibroadcast@drone.service: Scheduled restart job, restart counter is at 94.
Jan 14 23:25:13 raspberrypi systemd[1]: Stopped WFB profile drone.
Jan 14 23:25:13 raspberrypi systemd[1]: Started WFB profile drone.
Jan 14 23:25:14 raspberrypi wfb-server[2850]: [server.main #info] WFB version 23.1.14.32326-6251fc24
Jan 14 23:25:14 raspberrypi wfb-server[2850]: [server.main #info] Run on armv7l/5.4.51-v7l+ @raspberrypi, profile drone using wlan0
Jan 14 23:25:14 raspberrypi wfb-server[2850]: [server.main #info] Using cfg files:
Jan 14 23:25:14 raspberrypi wfb-server[2850]: /usr/lib/python3/dist-packages/wfb_ng/conf/master.cfg
Jan 14 23:25:14 raspberrypi wfb-server[2850]: /usr/lib/python3/dist-packages/wfb_ng/conf/site.cfg
Jan 14 23:25:14 raspberrypi wfb-server[2850]: /etc/wifibroadcast.cfg
Jan 14 23:25:14 raspberrypi wfb-server[2850]: [proxy._check_rc #info] # iw reg set BO
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [proxy._check_rc #info] # ip link set wlan0 down
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [proxy._check_rc #info] # iw dev wlan0 set monitor otherbss
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [proxy._check_rc #info] # ip link set wlan0 up
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [proxy._check_rc #info] # iw dev wlan0 set channel 1 HT20
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.init_mavlink #info] Open serial port ttyUSB0 on speed 115200
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.init_video #info] Listen for video stream 0 on 0.0.0.0:5602
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.TXProtocol.connectionMade #info] Started video tx
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.init_video #info] Video: /usr/bin/wfb_tx -p 0 -u 5602 -K /etc/drone.key -B 20 -G long -S 1 -L 1 -M 1 -k 8 -n 12 -T 0 -i 7669206 wlan0
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [udp.Port._bindSocket #info] UDPProxyProtocol starting on 14900
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [udp.Port._bindSocket #info] UDPProxyProtocol starting on 34349
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.init_tunnel #info] Tunnel RX: /usr/bin/wfb_rx -p 32 -u 14900 -K /etc/drone.key -i 7669206 wlan0
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.init_tunnel #info] Tunnel TX: /usr/bin/wfb_tx -p 160 -u 14901 -K /etc/drone.key -B 20 -G long -S 1 -L 1 -M 1 -k 1 -n 2 -T 0 -i 7669206 wlan0
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.RXProtocol.connectionMade #info] Started tunnel rx
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.TXProtocol.connectionMade #info] Started tunnel tx
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [common.abort_on_crash #info] Stopping reactor due to fatal error: [Errno 2] could not open port /dev/ttyUSB0: [Errno 2] No such file or directory: '/dev/ttyUSB0'
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.TXProtocol.processEnded #info] Stopped TX video tx with code 1
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [udp.Port.connectionLost #info] (UDP Port 14900 Closed)
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.TXProtocol.processEnded #info] Stopped TX tunnel tx with code 1
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [udp.Port.connectionLost #info] (UDP Port 34349 Closed)
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.RXProtocol.processEnded #info] Stopped RX tunnel rx with code None
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [base.EPollReactor.mainLoop #info] Main loop terminated.
Jan 14 23:25:15 raspberrypi wfb-server[2850]: [server.main #info] Exiting with code 1
Jan 14 23:25:15 raspberrypi systemd[1]: wifibroadcast@drone.service: Main process exited, code=exited, status=1/FAILURE
Jan 14 23:25:15 raspberrypi systemd[1]: wifibroadcast@drone.service: Failed with result 'exit-code'.
Jan 14 23:25:15 raspberrypi systemd[1]: Stopped WFB profile drone.

Apparetly, all the wifi chip stuff seems to be fine. But somehow, wifibroadcasts trys to open a nonexistend Serial port and crashes. Can anyone explain to me what this serial port does? Why do I need it? How can I turn it off?

**Comment by svpcom** [comment]
You can disable it `/etc/wifibroadcast.cfg`. Serial port used for telemetry link to autopilot. If you don't need autopilot telemetry just disable it
